src/main/java/uk/gov/companieshouse/accountsDates/AccountsDates.java

```diff
@@ -0,0 +1,199 @@
+		}
+	}
+
+	/**
+	 * @param periodStart
+	 * @param periodEnd
+	 * @param isSameYear
+	 * @return
+	 * 
+	 * 	Calculate balance sheet dates display format depending on range between
+	 *  period start and end dates
+	 * 
+	 */
+	public Map<String, String> calculatePeriodRange(LocalDate periodStart, LocalDate periodEnd, boolean isSameYear) {
+
+		int allowance = 15; // We allow year +/-15 days difference between dates to treat them as full year difference
+		int yearRangeMax = 365 + allowance;
+		int yearRangeMin = 365 - allowance;
+
+		Map<String, String> periodObject = new HashMap<>();
+
+		// period returns years, hours and days between dates eg 1 year, 3 months and 4 days
+		Period accountsPeriod = Period.between(periodStart, periodEnd);
+
+		long totalDaysDiff = ChronoUnit.DAYS.between(periodStart, periodEnd) + 1; // + 1 as they have time till the end of the day
+		long totalMonthsDiff = ChronoUnit.MONTHS.between(periodStart, periodEnd);
+
+		// If the days over an exact month are 1 to 15, then round down (unless thegperiod is less than 1 month, in which case round up).
+		// If the days over are 16 to 31, then round up. e.g. 15 months and 4 days is
+		// shown as 15 months; 15 months and 21 days is shown as 16 months.
+		totalMonthsDiff = (accountsPeriod.getDays() >= 15) ? totalMonthsDiff + 1 : totalMonthsDiff; // + 1 to days to include end date in calculation
+
+		// If the previous and current periods both end in the same year, then the
+		// heading is output as a full date e.g. ‘5 January 2015’ ’31 December 2015’.
+		if (isSameYear) {
+			periodObject.put(PERIOD_START, null);
+			periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
+		}
+
+		// If the accounting period is twelve months (+/- 15 days - period start to
+		// period end)
+		// Then the heading for the balance sheet figures is: ccyy e.g. 2015 2016
+		else if ((totalDaysDiff >= yearRangeMin && totalDaysDiff <= yearRangeMax) && totalMonthsDiff == 12) {
+
+//		else if ((accountsPeriod.getYears()==1 && (accountsPeriod.getDays()<=15)) || (accountsPeriod.getMonths()==11 && (accountsPeriod.getDays()>=15))){
+			periodObject.put(PERIOD_START, null);
+			DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy");
+			periodObject.put(PERIOD_END, periodEnd.format(formatter));
+		}
+
+		// If the accounting period is less than twelve months (<350 days - period start
+		// to period end) or
+		// more than twelve months (>380 days - period start to period end)
+		// Then the heading for the balance sheet figures is 'x' months to dd/mm/ccyy
+		// e.g. 15 months to 31/12/2016
+		else if (totalDaysDiff < yearRangeMin || totalDaysDiff > yearRangeMax) {
+			String monthsDiffString = (totalMonthsDiff == 1) ? " month" : " months";
+			periodObject.put(PERIOD_START, totalMonthsDiff + monthsDiffString);
+			periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
+		}
+
+		return periodObject;
```

Comment:
Sorry if I have read this wrong. What happens for this situation?
```
totalDaysDiff >= yearRangeMin && totalDaysDiff <= yearRangeMax && totalMonthsDiff != 12
```

An empty periodObject is being returned?

Reply:
this is the last else-if block of the balance sheet dates logic. 
periodObject.put(PERIOD_START, totalMonthsDiff + monthsDiffString);
periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));

the relevant periodObject is then returned

Reply:
resolved in 9cfc864